picklable_itertools/__init__.py

```diff
@@ -192,6 +192,73 @@ def _negated(self, argument):
         return not self._true_predicate(argument)
 
 
+class product(BaseItertool):
+    def __init__(self, *iterables):
+        self._iterables = [_iter(it) for it in iterables]
+        self._contents = [[] for it in iterables]
+        self._exhausted = [False for it in iterables]
+        self._position = [-1 for it in iterables]
+        self._initialized = False
+
+    def __next__(self):
+        # Welcome to the spaghetti factory.
+        def _next(i):
```

Comment:
We aren't storing the nested function anywhere, or ever passing it out of
scope, so I don't see how it could be a problem. I'll add the verify_pickle
though.